test/test_client.rb

```diff
@@ -103,12 +103,19 @@ def test_find_raises_on_unknown_quantity
   def test_find_retries_when_receiving_no_records_found
     @client.stubs(:find_every).raises(Rets::NoRecordsFound.new('')).then.returns([1])
 
-    assert_equal @client.find(:all), [1]
+    assert_equal [1], @client.find(:all)
   end
 
   def test_find_does_not_retry_when_receiving_no_records_found_with_option
     @client.stubs(:find_every).raises(Rets::NoRecordsFound.new('')).then.returns([1])
-    assert_equal @client.find(:all, no_records_not_an_error: true), []
+
+    assert_equal [], @client.find(:all, no_records_not_an_error: true)
+  end
+
+  def test_find_does_not_retry_and_returns_zero_on_count_request_when_receiving_no_records_found_with_option
+    @client.stubs(:find_every).raises(Rets::NoRecordsFound.new('')).then.returns([1])
```

Comment:
If I'm reading this correctly, we no longer need the `then.returns([1])`, right?